server/belga/io/feed_parsers/belga_dpa_newsml_2_0.py

```diff
@@ -175,14 +176,18 @@ def parse_content_meta(self, tree, item):
             if name_elt is not None:
                 item.setdefault('extra', {})['city'] = name_elt.text
 
-        # save all keywords as subject with scheme original-metadata
         for elem in meta.findall(self.qname('keyword')):
             data = elem.text.strip()
-            item.setdefault('subject', []).append({
-                'name': data,
-                'qcode': data,
-                'scheme': 'original-metadata'
-            })
+            belga_keywords = get_resource_service('vocabularies').get_items(_id='belga-keywords', qcode=data.upper())
+            if len(belga_keywords) > 0:
+                item.setdefault('subject', []).append(belga_keywords[0])
+            else:
+                # save all other unmapped keywords as subject with scheme original-metadata
+                item.setdefault('subject', []).append({
+                    'name': data,
+                    'qcode': data,
+                    'scheme': 'original-metadata'
+                })
```

Comment:
seems like this part is more or less duplicatedin base parser, can you avoid that duplication?